### DISCUSSION TITLE
HELP: It's not working for me

### DISCUSSION THREAD

**Discussion by relentless-coder** [mikesmithgh/kitty-scrollback.nvim#127] [Q&A] [answered]
Hello, I'm new to kitty and this plugin is exactly what I want, but I'm not able to make it work.
`:KittyScrollbackCheckHealth` is not showing any errors/warnings. However, running kitty_mod+h is not opening the pager.

```
font_family JetBrainsMono Nerd Font Mono SemiBold
font_size 13.0
enable_audio_bell no
map ctrl+option+t goto_layout tall
include ./theme.conf
allow_remote_control yes
listen_on unix:/tmp/kitty
shell_integration enabled

# kitty-scrollback.nvim Kitten alias
action_alias kitty_scrollback_nvim kitten /--Home--/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py
 
# Browse scrollback buffer in nvim
map kitty_mod+h kitty_scrollback_nvim
# Browse output of the last shell command in nvim
map kitty_mod+g kitty_scrollback_nvim --config ksb_builtin_last_cmd_output
# Show clicked command output in nvim
mouse_map ctrl+shift+right press ungrabbed combine : mouse_select_command_output : kitty_scrollback_nvim --config ksb_builtin_last_visited_cmd_output
```

**Comment by mikesmithgh** [comment]
@relentless-coder could you provide the output of `:KittyScrollbackCheckHealth` so that I can inspect if anything looks off?

What happens when you do `ctrl+shift+h`? Do you see any screen or just is it as if nothing happened at all?

**Reply by relentless-coder** [reply]
`:KittyScrollbackCheckHealth` is not giving any output. And running `kitty +kitten kitty_scrollback_nvim` gives `No kitten named kitty_scrollback_nvim`. `ctrl+shift+h` doesn't do anything. If I revert to default kitty conf then it opens the scrollback in `less`.

**Reply by mikesmithgh** [reply]
> And running kitty +kitten kitty_scrollback_nvim gives No kitten named kitty_scrollback_nvim

This suggests that your `action_alias` is not properly configured.

I see you wrote
```
action_alias kitty_scrollback_nvim kitten /--Home--/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py
```

Is `--Home--` actually replaced with the correct directory in your config? It should be the absolute path of `kitty_scrollback_nvim_.py`. Something similar to:
```
action_alias kitty_scrollback_nvim kitten /Users/mike/gitrepos/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py
```


The command `:KittyScrollbackGenerateKittens` should generate the correct path. (Try this if you have not already)


Could you let me know the full path that you have defined in your `action_alias`?

**Reply by relentless-coder** [reply]
Yes, the action_alias is set properly, don't know why I tried to hide my home path

`action_alias kitty_scrollback_nvim kitten /Users/relentless-coder/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py`

If I execute the python script then it gives following error:
```
Traceback (most recent call last):
  File "/Users/relentless-coder/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py", line 3, in <module>
    from kitty.boss import Boss
ModuleNotFoundError: No module named 'kitty'
```

I'm not sure if this is relevant.

**Reply by mikesmithgh** [reply]
odd, what happens if you try?
```
kitty @ kitten /Users/relentless-coder/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py
```

if that works, try
```
kitty @ kitten /Users/relentless-coder/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py --config ksb_builtin_checkhealth
```

**Reply by relentless-coder** [reply]
It didn't work

```
Traceback (most recent call last):
  File "kitty/boss.py", line 338, in _handle_remote_command
  File "kitty/remote_control.py", line 41, in handle_cmd
  File "kitty/rc/kitten.py", line 51, in response_from_kitty
  File "kitty/boss.py", line 877, in _run_kitten
  File "kittens/runner.py", line 52, in create_kitten_handler
  File "kittens/runner.py", line 34, in import_kitten_main_module
FileNotFoundError: [Errno 2] No such file or directory: '/Users/relentless_coder/.local/share/nvim/lazy/kitty_scrollback.nvim/python/kitty_scrollback_nvim.py'

[Errno 2] No such file or directory: '/Users/relentless_coder/.local/share/nvim/lazy/kitty_scrollback.nvim/python/kitty_scrollback_nvim.py'
```
But interesting thing is my username is `relentless-coder`, but somehow in the error stack it's changed to `relentless_coder`. I have confirmed that file exists at `/Users/relentless-coder/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py`

**Reply by mikesmithgh** [reply]
interesting, good catch that might be something. I know kitty prefers _ notation for its kittens. Maybe it is applying that to the full path? I can dig a bit on that.

**Reply by relentless-coder** [reply]
All right. Do let me know if I can help you in any way.

**Reply by mikesmithgh** [reply]
@relentless-coder what OS are you on?

**Reply by relentless-coder** [reply]
MacOS Ventura 13.6.3

**Reply by mikesmithgh** [reply, marked as answer]
thanks and what version of kitty? `kitty --version` ?

I am not able to reproduce on my side with 
```
kitty 0.31.0 created by Kovid Goyal
macOS 14.2.1 (23C71)
```

**Reply by relentless-coder** [reply]
That was the issue, I was using kitty 0.19. Upgraded kitty and now it's working as expected.

Thanks for your quick replies, and thank you for this wonderful plugin.

**Reply by mikesmithgh** [reply]
awesome, yes the min version of Kitty that can be used is `v0.29`. Glad it is working now 👍
